src/slider/__tests__/stateful-slider.test.js

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import React from 'react';
+import {mount} from 'enzyme';
+
+describe('Stateful slider', function() {
+  let allProps: any, wrapper;
+
+  beforeEach(function() {
+    allProps = {};
+    jest.mock('../slider', () => jest.fn(() => <div>test</div>));
```

Comment:
why do we need to mock the Slider here?

Reply:
we need to mock it right before we require it in test, so we are able to track mocks for each test. If we bubble require to `imports` jest will start mocking it for the whole file once and forever, so we can't know which test receives which mock call